CHATBOT ERROR HANDLING .More Informative and specific error displayed to the user.

Rahulsoni9321 opened the pull request:
Made Changes in the response that have been sent out by the server when It is receiving error from the Claude API. And made some changes on how error is handled on the frontend. 

### **Before**
![Screenshot 2024-11-22 201657](https://github.com/user-attachments/assets/ce07e890-67f9-4d05-b6c7-d38cb1e15576)
### _**After**_
![Screenshot 2024-11-22 201708](https://github.com/user-attachments/assets/61670214-96e7-4ec7-89de-2d0649149aa7)

I noticed that the chatbot in the repository is currently returning an error. Upon further investigation, it seems that the issue might be related to the API keys for Claude AI not being set up correctly. This could be why the chatbot isn't functioning as expected.

To resolve this, you may need to:
1. Verify that the API keys for Claude AI are correctly configured in the environment variables.

tyaga001 commented on src/app/api/summarize/route.ts:
```diff
@@ -30,10 +30,17 @@ export async function POST(request: Request) {
         return NextResponse.json({ answer });
     } catch (error: any) {
-        console.error('Error in API:', error.response?.data || error.message);
+        console.error(error.response?.data.error.message || error.message);
```
@coderabbitai nice tip - @Rahulsoni9321 what do you think?

tyaga001 commented on .env.example:
```diff
@@ -1,4 +1,5 @@
 DATABASE_URL=
 NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=
 CLERK_SECRET_KEY=
+ANTHROPIC_API_KEY=
```
I am thinking of open-source APIs to avoid unnecessary costs at this moment. I think Gemini have free or we can setup [GPT-3.5 Turbo](https://platform.openai.com/docs/models#gpt-3-5-turbo)

Rahulsoni9321 replied:
> I am thinking of open-source APIs to avoid unnecessary costs at this moment. I think Gemini have free or we can setup [GPT-3.5 Turbo](https://platform.openai.com/docs/models#gpt-3-5-turbo)

Makes Sense @tyaga001. Should I proceed with setting up the backend for the Gemini API and transitioning out of Claude, or is there a preferred direction you'd like to explore?

Rahulsoni9321 commented:
> @Rahulsoni9321 check this example and see if we can implement it in our blog - https://www.nikhilsnayak.dev/bot

Hey @tyaga001 Made a new commit where I had added the Gemini API setup and also the chat interface is changed completely. Al agent is only going to answer questions related to the blog and also did some major UI changes for better readability.
Formatted the response for better readability and abstracted the error message from the users that we are going to receive from the API providers. Only console logged the error response on the server for our reference.
Added a page which will be displayed if no conversation had happened between the bot and the user.

Rahulsoni9321 commented:
1. **RESPONSE READABILITY: BEFORE**

![Screenshot 2024-11-29 190248](https://github.com/user-attachments/assets/5dc9b98c-5a75-4958-a803-3a34945c1eaa)

2. **RESPONSE READABILITY:  AFTER**
 
![Screenshot 2024-11-30 000408](https://github.com/user-attachments/assets/bc9b734a-e9d4-4c57-9661-77975c58261c)


3.  **INTERFACE BEFORE ANY CONVERSATION BETWEEN AI AND USER**

![Screenshot 2024-11-30 000342](https://github.com/user-attachments/assets/215149ef-21da-4e91-adc4-26f8e47581e9)

4. **BUTTON CHANGED**

![Screenshot 2024-11-30 001456](https://github.com/user-attachments/assets/3d6dca45-fce0-4974-954a-d1c9e421154b)

5. **WHEN QUESTION IS ASKED OUTSIDE THE BLOG CONTENT**
6. 
![Screenshot 2024-11-30 001538](https://github.com/user-attachments/assets/b4d01e4b-c415-4538-bf4e-af59e1167f8f)

tyaga001 commented on package.json:
```diff
@@ -23,6 +24,7 @@
     "@radix-ui/react-dropdown-menu": "^2.1.1",
     "@radix-ui/react-slot": "^1.1.0",
     "@radix-ui/react-tabs": "^1.1.0",
+    "@radix-ui/react-toast": "^1.2.2",
```
@Rahulsoni9321 check this pls let's just keep @radix-ui/react-toast

tyaga001 commented on package.json:
```diff
@@ -45,8 +47,10 @@
+    "sonner": "^1.7.0",
     "tailwind-merge": "^2.4.0",
     "tailwindcss-animate": "^1.0.7",
+    "ui": "github:shadcn/ui",
```
@Rahulsoni9321 check this one as well

Rahulsoni9321 commented on src/app/api/summarize/route.ts:
```diff
@@ -1,38 +1,42 @@
+import { NextResponse } from "next/server";
+
+
+const GEMINI_API_KEY = process.env.GEMINI_API_KEY;
```
> @Rahulsoni9321 pls check all the suggestions raised by @coderabbitai

@tyaga001  made all the suggested changes by @coderabbitai and open a new pr (#49), Please have a look and let me know if changes needed.

tyaga001 commented:
@nirnejak pls check do we need this change anymore if yes let's fix it.
